### PR TITLE
Add Effective Dart analysis options list

### PR DESCRIPTION
**Pull request by tenhobi**
As is described in #288, I have tried to set a list of options describing Effective Dart analysis options. I have also included (in comments) unuses options so it will be simplier to see what can be added / what is missing.

All rules are sorted in sections and subsections according to Effective Dart, so it will be easier to sync the rules in time.

### CODE REVIEW

File: example/effective-dart.yaml [hunk 1]
```diff
@@ -0,0 +1,160 @@
+# List of rules following the Effective Dart guide.
+# https://www.dartlang.org/guides/language/effective-dart
+
```
Comment (pq) [inline example/effective-dart.yaml]: Can we call out the incompleteness / in-progress-ness of this list?  Similar to what we're doing here:

https://github.com/dart-lang/linter/blob/97a6ec9e061c9bd0c0d85f7587f78d856ecb8ece/example/google.yaml#L7

(Possibly also in the `README`?)

Comment (tenhobi) [reply example/effective-dart.yaml]: Sure, I will add the notes.

**Comment by tenhobi** [comment]
@pq I have updated the notes.

File: README.md [hunk 2]
```diff
@@ -32,7 +32,7 @@ you could lint your package like this:
 
     $ dartanalyzer --options analysis_options.yaml .
     
-and see any violations of the `annotate_overrides`, `hash_and_equals`, and `prefer_is_not_empty` rules in the console.  To help you choose the rules you want to enable for your package, we have provided both a [complete list of rules][lints] and a list of the [lints that are enforced internally at Google][google-lints].
+and see any violations of the `annotate_overrides`, `hash_and_equals`, and `prefer_is_not_empty` rules in the console.  To help you choose the rules you want to enable for your package, we have provided a [complete list of rules][lints], an up-to-date list of the [lints according to the Effective Dart guide][effective-dart-lints] and a list of the [lints that are enforced internally at Google][google-lints].
```
Comment (pq) [inline README.md]: I wonder if this doesn't suggest that the list is complete...  What about something like:

> a growing list of [lints derived from the Effective Dart guide]

or

> an evolving list of [lints that are enforce advice in the Effective Dart guide]

@bwilkerson : you're good at word-smithing... Any ideas?

Comment (bwilkerson) [reply README.md]: In addition to making it clear that the rule set will be expanded as new rules are added, we probably need to be clear that not everything suggested in the guide has a rule associated with it (and hence the rule set will always be incomplete in some sense).

**Comment by pq** [comment]
Super.  Thanks @tenhobi!

**Pull request merged**